cli/src/dap_client.rs

```diff
@@ -0,0 +1,170 @@
+                            Prio3SumVec::new_sum_vec(
+                                2,
+                                bits as usize,
+                                length as usize,
+                                chunk_length as usize,
+                            )?,
+                        )
+                        .await?;
+                        let v = measurement
+                            .split(',')
+                            .map(|s| s.trim().parse())
+                            .collect::<Result<_, _>>()
+                            .map_err(|e| {
+                                Error::Other(format!("cannot parse measurement as number: {e:?}"))
+                            })?;
+                        client.upload(&v).await?;
+                    }
+                    divviup_client::Vdaf::Histogram(Histogram::Categorical {
+                        buckets,
+                        chunk_length,
+                    }) => {
+                        // Chunk length should always be set for DAP-09 tasks
+                        let chunk_length = chunk_length.ok_or_else(|| {
+                            Error::Other(
+                                "task's VDAF configuration does not provide a chunk length".into(),
+                            )
+                        })?;
+                        let client = janus_client::Client::new(
+                            task_id.parse()?,
+                            leader_url,
+                            helper_url,
+                            Duration::from_seconds(task.time_precision_seconds as u64),
+                            Prio3Histogram::new_histogram(2, buckets.len(), chunk_length as usize)?,
+                        )
+                        .await?;
+                        let v = measurement.parse().map_err(|e| {
+                            Error::Other(format!("cannot parse measurement as number: {e:?}"))
+                        })?;
+                        client.upload(&v).await?;
+                    }
+                    divviup_client::Vdaf::Histogram(Histogram::Continuous {
+                        buckets,
+                        chunk_length,
+                    }) => {
+                        // Chunk length should always be set for DAP-09 tasks
+                        let chunk_length = chunk_length.ok_or_else(|| {
+                            Error::Other(
+                                "task's VDAF configuration does not provide a chunk length".into(),
+                            )
+                        })?;
+                        let client = janus_client::Client::new(
+                            task_id.parse()?,
+                            leader_url,
+                            helper_url,
+                            Duration::from_seconds(task.time_precision_seconds as u64),
+                            Prio3Histogram::new_histogram(2, buckets.len(), chunk_length as usize)?,
```

Comment:
We need an additional bucket for the uppermost boundary to infinity.
```suggestion
                            Prio3Histogram::new_histogram(2, buckets.len() + 1, chunk_length as usize)?,
```